mapeditor/mapsettings/timedevent.cpp

```diff
@@ -89,10 +98,63 @@ void TimedEvent::on_TimedEvent_finished(int result)
 	}
 	descriptor["resources"] = res;
 
+	QVariantList deletedObjects;
+	for(int i = 0; i < ui->deletedObjects->count(); ++i)
+	{
+		auto const & item = ui->deletedObjects->item(i);
+		auto data = item->data(MapEditorRoles::ObjectInstanceIDRole);
+		auto id = ObjectInstanceID(data.value<int>());
+		deletedObjects.push_back(QVariant::fromValue(id.num));
+	}
+	descriptor["deletedObjectsInstances"] = QVariant::fromValue(deletedObjects);
+
 	target->setData(Qt::UserRole, descriptor);
 	target->setText(ui->eventNameText->text());
 }
 
+void TimedEvent::on_addObjectToDelete_clicked()
+{
+	for(int lvl : {0, 1})
+	{
+		auto & l = controller.scene(lvl)->objectPickerView;
+		l.highlight<const CGObjectInstance>();
+		l.update();
+		QObject::connect(&l, &ObjectPickerLayer::selectionMade, this, &TimedEvent::onObjectPicked);
+	}
+	hide();
+	dynamic_cast<QWidget *>(parent()->parent()->parent()->parent()->parent()->parent()->parent())->hide();
+}
+
+void TimedEvent::on_removeObjectToDelete_clicked()
+{
+	delete ui->deletedObjects->takeItem(ui->deletedObjects->currentRow());
+}
+
+void TimedEvent::onObjectPicked(const CGObjectInstance * obj)
+{
+	show();
+	dynamic_cast<QWidget *>(parent()->parent()->parent()->parent()->parent()->parent()->parent())->show();
```

Comment:
wtf wtf wtf wtf wtf wtf wtf is this?

Reply:
not my code :rofl:

Reply:
@godric3 then?
In any case - can this be fixed or at least explained?

Reply:
ah yes, guilty, copied from victory conditions (https://github.com/vcmi/vcmi/blob/develop/mapeditor/mapsettings/victoryconditions.cpp#L536)
If I remember correctly it was required to hide/show map setting dialog, to properly show map with selection layer/return to event settings. 
I don't know qt well enough to have it fixed quickly so I used same approach as found in codebase, I may look more into in on the weekend but can't promise anything